ethToElrond/v2/elrondToEth/steps/step01GetPending.go

```diff
@@ -0,0 +1,61 @@
+package steps
+
+import (
+	"context"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/core"
+	"github.com/ElrondNetwork/elrond-eth-bridge/ethToElrond/v2/elrondToEth"
+)
+
+type getPendingStep struct {
+	bridge elrondToEth.ElrondToEthBridge
+}
+
+func (step *getPendingStep) Execute(ctx context.Context) (core.StepIdentifier, error) {
+	step.bridge.ResetRetriesCountOnEthereum()
+	step.bridge.ResetRetriesCountOnElrond()
+
+	batch, err := step.bridge.GetBatchFromElrond(ctx)
```

Comment:
why not using only one action GetAndStoreBatchFromElrond and then only check err for nil and batch to nil?

Reply:
nevermind, found the answer in the following steps